src/Canvas.js

```diff
@@ -168,7 +168,7 @@ var Canvas = React.createClass({
                         || nextProps.columns !== this.props.columns
                         || nextProps.width !== this.props.width
                         || nextProps.cellMetaData !== this.props.cellMetaData
-                        || !shallowEqual(nextProps.style, this.props.style);
+                        || shallowCloneObject(nextProps.style) !== shallowCloneObject(this.props.style);
```

Comment:
thats right. shallowCloneObject is not the right thing to use here, we want to compare the objects not copy them. We should just be checking each property of incoming props and check if equal to the current props. There is a `src/addons/utils` folder available. I think we just copy any utility functions in there rather than relying on external files from `react` package